devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+    return resultMap
+}
+
+def setheatingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setcoolingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setSetpoint(Integer attrId, Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    
+	def cmds = zigbee.writeAttribute(0x0201, attrId , 0x29 , zigBeeTemp) 
+}
+
+def setHeatingSetpoint(Double degrees) {
+	
+	sendEvent(name: "heatingSetpoint", value: degrees)
+	setSetpoint(0x12, degrees)
+}
+
+def setCoolingSetpoint(Double degreesF) {
+
+	sendEvent(name: "coolingSetpoint", value: degreesF)
+	setSetpoint(0x11, degreesF)
+}
+
+def setCoolLimit(Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    // minmum cool limit
+	[ zigbee.writeAttribute(0x0201, 0x0017 , 0x29 , zigBeeTemp) ]
+}
+
+def setHeatLimit(Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    // maximum heat limit
+	[ zigbee.writeAttribute(0x0201, 0x0016 , 0x29 , zigBeeTemp) ]
+}
+
+def setThermostatMode(String next) {
+	def val = (getModeMap().find { it.value == next }?.key)?: "00"
+	def mode = device.currentState("thermostatMode")?.value
+	
+	sendEvent("name":"thermostatMode", "value":"$next")
+	
+	[ zigbee.writeAttribute(0x0201, 0x1C , 0x30 , val) + 
+		"delay 300" +
+	  zigbee.readAttribute(0x0201, 0x1C) 
+	]
+}
+
+def setThermostatFanMode(String value) {
+
+	sendEvent(name: "thermostatFanMode", value: value)
+	def val = (getFanModeMap().find { it.value == value }?.key)?: "00"
+	[ zigbee.writeAttribute(0x0201, 0x10 , 0x30 , val, ["mfgCode": 0x1121]) + 
+	  "delay 300" +
+	  zigbee.readAttribute(0x0201, 0x10, ["mfgCode": 0x1121]) ]
+}
+
+def off() { setThermostatMode("off") }
+def heat() { setThermostatMode("heat") }
+def auto() { setThermostatMode("auto") }
+def cool() { setThermostatMode("cool") }
+
+def fanOn() { setThermostatFanMode("fanOn") }
+def fanAuto() { setThermostatFanMode("fanAuto") }
+def fanCirculate() { setThermostatFanMode("fanCirculate") }
+
+def tempUp()   { adjustSetpoint(5, "")      }
+def tempDown() { adjustSetpoint(-5, "")     }
+def heatUp()   { adjustSetpoint(5, "heat")  }
+def heatDown() { adjustSetpoint(-5, "heat") }
+def coolUp()   { adjustSetpoint(5, "cool")  }
+def coolDown() { adjustSetpoint(-5, "cool") }
+
+private adjustSetpoint(Number value, String mode) {
+		
+	if ("cool" != mode && "heat" != mode){
+		mode = device.currentState("thermostatMode")?.value
+		if (mode == "auto")
+			mode = device.currentState("runningMode")?.value
+	}
+	
+	//POSSIBLE ST BUG: ST platform seems to transmit the raiseLowerTemperature command 3 times over the air causing the setpoint to be 3x higher
+	//  Instead of using the zigbee command for raising or lower setpoint, we'll fiddle with the setpoint and manually adjust
+
+	//default to both heat and cool
+    Integer attrId
+	Number currentSetPoint
+    Integer zigBeeTemp
+	String attrName 
+	if ("heat" == mode ){
+    	attrId = 0x12
+		currentSetPoint = device.currentValue("heatingSetpoint")
```

Comment:
Consider that current state may be in either °F or °C depending on the last reported value, provided the unit used is matching the current location scale, which it should be in order to keep the displayed temperature, the Recently tab records and saved state in line with the current location scale.
See Zen Thermostat as reference.